docs/cudf/source/developer_guide/documentation.md

```diff
@@ -0,0 +1,199 @@
+# Writing documentation
+
+cuDF documentation is split into multiple pieces.
+All core functionality is documented using inline docstrings.
+Additional pages like user or developer guides are written independently.
+While docstrings are written using [reStructuredText](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html) (reST),
+the latter are written using [MyST](https://myst-parser.readthedocs.io/en/latest/) 
+The inline docstrings are organized using a small set of additional reST pages.
+The results are all then compiled together using [Sphinx](https://www.sphinx-doc.org/en/master/).
+This document discusses each of these components and how to contribute to them.
+
+## Docstrings
+
+cuDF docstrings use the [numpy](https://numpydoc.readthedocs.io/en/latest/format.html) style.
+In lieu of a complete explanation,
+we include here an example of the format and the commonly used sections:
+
+```
+class A:
+    """Brief description of A.
+
+    Longer description of A.
+
+    Parameters
+    ----------
+    x : int
+        Description of x, the first constructor parameter.
+    """
+    def __init__(self, x: int):
+        pass
+
+    def foo(self, bar: str):
+        """Short description of foo.
+
+        Longer description of foo.
+
+        Parameters
+        ----------
+        bar : str
+            Description of bar.
+        
+        Returns
+        -------
+        float
+            Description of the return value of foo.
+
+        Raises
+        ------
+        ValueError
+            Explanation of when a ValueError is raised.
+            In this case, a ValueError is raised if bar is "fail".
+
+        Examples
+        --------
+        The examples section is _strongly_ encouraged.
+        Where appropriate, it may mimic the examples for the corresponding pandas API.
+        >>> a = A()
+        >>> a.foo('baz')
+        0.0
+        >>> a.foo('fail')
+        ...
+        ValueError: Failed!
+        """
+        if bar == "fail":
+            raise ValueError("Failed!")
+        return 0.0
+```
+
+`numpydoc` supports a number of other sections of docstrings.
+Developers should familiarize themselves with them, since many are useful in different scenarios.
+Our guidelines include one addition to the standard the `numpydoc` guide.
+Class properties, which are not explicitly covered, should be documented in the getter function.
+That choice makes `help` more useful as well as enabling docstring inheritance in subclasses.
+
+All of our docstrings are validated using [`pydocstyle`](http://www.pydocstyle.org/en/stable/).
```

Comment:
Are `numpydoc`'s validations also run? https://numpydoc.readthedocs.io/en/latest/validation.html

Reply:
Oh wow no, I had no idea that those even existed. That looks extremely useful. I added that to the checklist in #9999

Reply:
#11253